src/libstd/collections/hash/table.rs

```diff
@@ -38,23 +37,22 @@ const EMPTY_BUCKET: u64 = 0u64;
 ///
 /// Essential invariants of this structure:
 ///
-///   - if t.hashes[i] == EMPTY_BUCKET, then `Bucket::at_index(&t, i).raw`
+///   - if t.hashes[i] == None, then `Bucket::at_index(&t, i).raw`
 ///     points to 'undefined' contents. Don't read from it. This invariant is
 ///     enforced outside this module with the `EmptyBucket`, `FullBucket`,
 ///     and `SafeHash` types.
 ///
 ///   - An `EmptyBucket` is only constructed at an index with
-///     a hash of EMPTY_BUCKET.
+///     a hash of None.
 ///
-///   - A `FullBucket` is only constructed at an index with a
-///     non-EMPTY_BUCKET hash.
+///   - A `FullBucket` is only constructed at an index with a hash.
 ///
-///   - A `SafeHash` is only constructed for non-`EMPTY_BUCKET` hash. We get
+///   - A `SafeHash` is only constructed for non-zero hash. We get
 ///     around hashes of zero by changing them to 0x8000_0000_0000_0000,
 ///     which will likely map to the same bucket, while not being confused
 ///     with "empty".
 ///
-///   - All three "arrays represented by pointers" are the same length:
+///   - Both "arrays represented by pointers" have the same length:
 ///     `capacity`. This is set at creation and never changes. The arrays
 ///     are unzipped to save space (we don't have to pay for the padding
 ///     between odd sized elements, such as in a map from u64 to u8), and
```

Comment:
This is no longer true - the keys and values are mixed together, so we do have to pay for the extra padding.